File: src/main/java/net/devium/argentum/rest/model/OrderItemRequest.java
package net.devium.argentum.rest.model;

public class OrderItemRequest {
    private long productId;
    private int quantity;

    public OrderItemRequest() {
    }

    public OrderItemRequest(long productId, int quantity) {
        this.productId = productId;
        this.quantity = quantity;
    }

    public long getProductId() {
        return productId;
    }

    public int getQuantity() {
        return quantity;
    }
}

File: argentum-ui/src/app/common/rest-service/request/order-item-request.ts
import { Product } from '../../model/product';
export class OrderItemRequest {
  productId: number;
  quantity: number;
}

export function fromOrderItem(entry: [Product, number]): OrderItemRequest {
  return {
    productId: entry[0].id,
    quantity: entry[1]
  };
}
